sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -321,6 +321,16 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     // bucket specification to empty. Note that partition columns are retained, so that we can
     // call partition-related Hive API later.
     def newSparkSQLSpecificMetastoreTable(): CatalogTable = {
+      if (table.partitionColumnNames.contains(EMPTY_DATA_SCHEMA.head.name)) {
+        // This is a valid use case but for historical reasons we don't allow it as the 'col' name
+        // is taken by the empty schema. We should allow this in the future if we find a proper
```

Comment:
If we should allow this, we had better add IDed TODO here, @yaooqinn .

Reply:
Filed SPARK-51957, thank you @dongjoon-hyun